webui/react/src/components/MetricSelect.tsx

```diff
@@ -149,37 +143,14 @@ const MetricSelectFilter: React.FC<Props> = ({
     );
   }, [totalNumMetrics, visibleMetrics]);
 
-  const [maxTagCount, selectorPlaceholder] = useMemo(() => {
-    // This should never happen, but fall back to inoffensive empty placeholder
-    if (metricValues === undefined) {
-      return [0, ''];
-    }
-    if (metricValues.length === 0) {
-      // If we set maxTagCount=0 in this case, this placeholder will not be displayed.
-      return [-1, 'None selected'];
-    } else if (metricValues.length === totalNumMetrics) {
```

Comment:
I wanted to call this line out specifically since it is a change in behavior. In this specific implementation we add `2` extra options to `Reset to Default` and select `All` however, when every metric in the list is selected, we will show the text `5 selected` for example instead of `All` since not all of the possible dropdown values are selected. 
 
 This logic and behavior is technically correct so in my opinion there is nothing to be changed here. 
 
 Just a call out since this is different behavior than today.

Reply:
@keita-determined this semi-related to your last comment concerning total options, thought I would just let it be aware. In my opinion it is reasonable behavior.